packages/router/router.ts

```diff
@@ -2624,16 +2628,22 @@ async function callLoaderOrAction(
       );
 
       // Check if this an external redirect that goes to a new origin
-      let external = createURL(location).origin !== createURL("/").origin;
+      let currentUrl = new URL(request.url);
+      let currentOrigin = currentUrl.origin;
+      let newOrigin = new URL(location, currentOrigin).origin;
+      let external = newOrigin !== currentOrigin;
```

Comment:
This is the main fix for cloudflare - we inline the URL creation and just use the current request to determine the origin